### ISSUE TITLE
Fix format_operands in instructions

### ISSUE THREAD

**Issue by reox** [androguard/androguard#656]
Similar to #254:

Android stores only integer values in the instructions (see https://source.android.com/devices/tech/dalvik/instruction-formats.html) for constants.
But there are obviously floats and other types in java.
Thus, there is some conversion going on at the time the variable is used.
That means during disassembly time, we do not know really what the type of the constant value will be. So we should use the type and value which is defined in the documentation (https://source.android.com/devices/tech/dalvik/dalvik-bytecode)
Then we need some function which can construct the correct value - which can only be created using the decompiler to decide the type of the constant.

In the test case, this discrepancy was visible, as it would not work. Using some formatting of the value, it is possible to convert it:
https://github.com/androguard/androguard/blob/5bb68c3a5b8301217164bddc519cd319a72d27ae/tests/test_types.py#L140-L165
But I'm not sure if this is all correct...

Long story short: the formatted_operands are probably not valid. Better idea would be to replace it with a system where you specify the type you want and the conversion is made.

**Comment by reox** [comment]
removed it.... ;)
